src/main/kotlin/calculator/PositiveNumber.kt

```diff
@@ -0,0 +1,16 @@
+package calculator
+
+data class PositiveNumber(val value: Int) {
+
+    init {
+        require(value > 0) { "양수만 허용합니다. value: $value" }
+    }
+
+    constructor(value: String) : this(
+        value.toIntOrNull() ?: throw IllegalArgumentException("허용하지 않는 문자열입니다. value: $value")
+    )
+
+    fun plus(that: PositiveNumber): PositiveNumber {
```

Comment:
코틀린에서는 정해진 이름의 함수를 오버로딩함으로써 표준 수학 연산자를 오버로딩할 수 있습니다. 아래의 글이 도움 될 거예요.
https://kotlinlang.org/docs/operator-overloading.html

```suggestion
    operator fun plus(that: PositiveNumber): PositiveNumber {
```